### PR TITLE
API-10300: new tags management

### PR DESCRIPTION
**Pull request by grz-gajda**

### CODE REVIEW

File: src/configuration/index.ts [hunk 1]
```diff
@@ -452,4 +453,38 @@ export default class ConfigurationAPI extends WebAPI {
...
     return this.send("list_channels", {});
   }
+
+  /**
+   * Creates a new tag.
+   * @param name - name of the created tag
+   * @param groupIDs - list of groups where tag will be assigned
+   */
+  async createTag(name: string, groupIDs: number[]): Promise<void> {
+    await this.send("list_tags", { name, group_ids: groupIDs });
```
Comment (janowsiany) [inline src/configuration/index.ts:463]:
```suggestion
    await this.send("create_tag", { name, group_ids: groupIDs });
```

File: src/configuration/index.ts [hunk 2]
```diff
@@ -452,4 +453,38 @@ export default class ConfigurationAPI extends WebAPI {
...
+  }
+
+  /**
+   * Deletes an existing tag.
+   * @param name - name of the removed tag
```
Comment (janowsiany) [inline src/configuration/index.ts:468]:
```suggestion
   * @param name - name of the tag to be deleted
```

File: src/configuration/index.ts [hunk 3]
```diff
@@ -452,4 +453,38 @@ export default class ConfigurationAPI extends WebAPI {
...
+  async listTags(groupIDs: number[]): Promise<Tag[]> {
+    return this.send("list_tags", { group_ids: groupIDs });
+  }
+
+  /**
+   * Updates an existing tag.
+   * @param name - name of the updated tag
+   * @param groupIDs - list of groups where tag will be assigned
+   */
+  async updateTag(name: string, groupIDs: number[]): Promise<void> {
```
Comment (janowsiany) [inline src/configuration/index.ts:487]:
```suggestion
  async updateTag(name: string, groupIDs?: number[]): Promise<void> {
```

File: src/configuration/index.ts [hunk 4]
```diff
@@ -452,4 +453,38 @@ export default class ConfigurationAPI extends WebAPI {
...
+   */
+  async deleteTag(name: string): Promise<void> {
+    await this.send("delete_tag", { name });
+  }
+
+  /**
+   * Returns tags assigned to requested groups.
+   * @param groupIDs - filter tags by groups where there are assigned
+   */
+  async listTags(groupIDs: number[]): Promise<Tag[]> {
```
Comment (janowsiany) [inline src/configuration/index.ts:478]:
```suggestion
  async listTags(groupIDs?: number[]): Promise<Tag[]> {
```

File: src/configuration/index.ts [hunk 5]
```diff
@@ -452,4 +453,38 @@ export default class ConfigurationAPI extends WebAPI {
   async listChannels(): Promise<ChannelActivity[]> {
     return this.send("list_channels", {});
   }
+
+  /**
+   * Creates a new tag.
+   * @param name - name of the created tag
+   * @param groupIDs - list of groups where tag will be assigned
+   */
+  async createTag(name: string, groupIDs: number[]): Promise<void> {
```
Comment (janowsiany) [inline src/configuration/index.ts:462]:
```suggestion
  async createTag(name: string, groupIDs:? number[]): Promise<void> {
```

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: src/configuration/index.ts (hunk 1 above)
Comment (grz-gajda) [reply src/configuration/index.ts]: 😞  what a mistake.

File: src/configuration/index.ts (hunk 2 above)
Comment (grz-gajda) [reply src/configuration/index.ts]: Applied.

File: src/configuration/index.ts (hunk 4 above)
Comment (grz-gajda) [reply src/configuration/index.ts]: Applied.

**Review by janowsiany — APPROVED** [review]

**Pull request merged**